sdk/tests/utilities/test_token_refresh.py

```diff
@@ -139,17 +122,29 @@ def test_refreshed_token_when_expired(self):
 
         sleep(1)
 
-        self.assertNotEqual(first_value, refreshed_token)
+        with patch("requests.post", side_effect=[
+            self.MockApiResponse(
+                json_data={
+                    "error": "invalid_grant",
+                    "error_description": "The refresh token is invalid or expired."
+                },
+                status_code=400
+            ),
+            self.MockApiResponse(
+                json_data={
+                    "access_token": "mock_access_token",
+                    "refresh_token": "mock_refresh_token",
+                    "expires_in": 60
+                },
+                status_code=200
+            ),
+        ]) as mock_requests:
```

Comment:
This creates a sequence of mocked responses to simulate an expired refresh token.

1) Returns a 400 code with the refresh token invalid error
2) On the next call returns a 200 with the mock_access_token and mock_refresh_token.